src/core/network/injectSendNetworkRequest.js

```diff
@@ -43,11 +44,21 @@ export default ({
         : sendFetchRequest;
 
       return requestMethod(url, stringifiedPayload).then(response => {
-        if (
-          isRetryableHttpStatusCode(response.status) &&
-          retriesAttempted < 3
-        ) {
-          return executeRequest(retriesAttempted + 1);
+        const requestIsRetryable = isRequestRetryable({
+          response,
+          retriesAttempted
+        });
+
+        if (requestIsRetryable) {
+          const requestRetryDelay = getRequestRetryDelay({
+            response,
+            retriesAttempted
+          });
+          return new Promise(resolve => {
+            setTimeout(() => {
+              resolve(executeRequest(retriesAttempted + 1));
+            }, requestRetryDelay);
```

Comment:
Previously, we were retrying the request right away. Now we have an incremental, moderately randomized backoff.

Reply:
Nice!

Reply:
Just want to double check this: when retrying, is the same `requestId` being used or a new one would be generated?

Reply:
The request ID will be re-used. Would you expect a new ID to be generated?